modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -801,11 +846,14 @@ private void acknowledgeFragment(String nodeName, long fragmentId, @Nullable Thr
 
                 if (rootFragmentId0 != null && fragmentId == rootFragmentId0) {
                     root.completeExceptionally(ex);
+                } else if (root == null) {
```

Comment:
This doesn't help me a lot, because in this case we fail on the last line of this method
```
remoteFragmentInitCompletion.get(new RemoteFragmentKey(nodeName, fragmentId)).complete(null);
```

I have almost 100% NPE when run `ItSqlAsynchronousApiTest.testQueryTimeout()`, and I'm sure that the root issue is not here (at least I seee that first NPE happen in executeFragment method) .

Since we don't observe any issues when running tests on TeamCity - let's just revert this change (if (root == null)) and investigate this separately?

Reply:
Fixed NPE.

Reply:
Filed an issue 
https://issues.apache.org/jira/browse/IGNITE-22585